documentation/bolt_project_directories.md

```diff
@@ -4,6 +4,16 @@ Bolt runs in the context of a project directory or a `Boltdir`. This directory c
 
 The project directory structure makes it easy to share Bolt code by committing the project directory to Git. You can then check different repositories of Bolt code into different directories in order to manage various applications.
 
+## Project directory names
+
+Project directory names must match the expression: `[a-z][a-z0-9_]*`. In other words, they
+can contain only lowercase letters, numbers, and underscores, and begin with a 
+lowercase letter.
+
+Avoid giving your project the same name as another module in your modulepath.
```

Comment:
Would it be useful to say why users should avoid this? I'm not sure if there's a real use case for wanting to shadow an installed module, but since Bolt won't warn/break if you do it might be helpful to explain what would happen.

Reply:
The use case for wanting to shadow an installed module is developing a module that's installed elsewhere on the modulepath. Agree it'd be better to say "Projects will take precedence over installed modules of the same name, and will error if they conflict with builtin Bolt modules".